packages/components/src/components/dialog/dialog.component.ts

```diff
@@ -0,0 +1,362 @@
+   * use the `show()` and `hide()` methods and this attribute will reflect the dialog's open state.
+   */
+  @property({ type: Boolean, reflect: true }) open = false;
+
+  /**
+   * The dialog's label as displayed in the header. You should always include a relevant label even when using
+   * `no-header`, as it is required for proper accessibility. If you need to display HTML, use the `label` slot instead.
+   */
+  @property({ reflect: true }) label = '';
+
+  /**
+   * Disables the header. This will also remove the default close button, so please ensure you provide an easy,
+   * accessible way for users to dismiss the dialog.
+   */
+  @property({ attribute: 'no-header', type: Boolean, reflect: true }) noHeader = false;
+
+  firstUpdated() {
+    this.dialog.hidden = !this.open;
+
+    if (this.open) {
+      this.addOpenListeners();
+      this.modal.activate();
+      lockBodyScrolling(this);
+    }
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.modal.deactivate();
+    unlockBodyScrolling(this);
+    this.closeWatcher?.destroy();
+  }
+
+  private requestClose(source: 'close-button' | 'keyboard' | 'overlay') {
+    const slRequestClose = this.emit('syn-request-close', {
+      cancelable: true,
+      detail: { source }
+    });
+
+    if (slRequestClose.defaultPrevented) {
+      const animation = getAnimation(this, 'dialog.denyClose', { dir: this.localize.dir() });
+      animateTo(this.panel, animation.keyframes, animation.options);
+      return;
+    }
+
+    this.hide();
+  }
+
+  private addOpenListeners() {
+    if ('CloseWatcher' in window) {
+      this.closeWatcher?.destroy();
+      this.closeWatcher = new CloseWatcher();
+      this.closeWatcher.onclose = () => this.requestClose('keyboard');
+    } else {
+      document.addEventListener('keydown', this.handleDocumentKeyDown);
+    }
+  }
+
+  private removeOpenListeners() {
+    this.closeWatcher?.destroy();
+    document.removeEventListener('keydown', this.handleDocumentKeyDown);
+  }
+
+  private handleDocumentKeyDown = (event: KeyboardEvent) => {
+    if (event.key === 'Escape' && this.modal.isActive() && this.open) {
+      event.stopPropagation();
+      this.requestClose('keyboard');
+    }
+  };
+
+  @watch('open', { waitUntilFirstUpdate: true })
+  async handleOpenChange() {
+    if (this.open) {
+      // Show
+      this.emit('syn-show');
+      this.addOpenListeners();
+      this.originalTrigger = document.activeElement as HTMLElement;
+      this.modal.activate();
+
+      lockBodyScrolling(this);
+
+      // When the dialog is shown, Safari will attempt to set focus on whatever element has autofocus. This can cause
+      // the dialogs's animation to jitter (if it starts offscreen), so we'll temporarily remove the attribute, call
+      // `focus({ preventScroll: true })` ourselves, and add the attribute back afterwards.
+      //
+      // Related: https://github.com/synergy-design-system/synergy/issues/693
```

Comment:
Just a note on something we should look at at some point:
Here and on other components (e.g. select, input, ...) we have a link to an github issue. But this link is not correct, because we overwrite all shoelace occurrences. Should be something `https.//github.com/shoelace/shoelace....`

Maybe we need to overwrite it with vendorism again in all components, where we have this problem. Or we'll just ignore it.
What do you think?

Reply:
I have created a ticket for this at https://github.com/synergy-design-system/synergy-design-system/issues/471.